integration-test/src/test/java/com/datastrato/gravitino/integration/test/util/spark/SparkTableInfo.java

```diff
@@ -0,0 +1,143 @@
+/*
+ *  Copyright 2024 Datastrato Pvt Ltd.
+ *  This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.integration.test.util.spark;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import lombok.Data;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.jupiter.api.Assertions;
+
+/** The output of SparkSQL is parsed to SparkTableInfo to check the result in test. */
+@Data
+public class SparkTableInfo {
+  private String name;
+  private String database;
+  private String comment;
+  private List<SparkColumnInfo> columns = new ArrayList<>();
+  private Map<String, String> tableProperties;
+  private List<String> unknownItems = new ArrayList<>();
+
+  public SparkTableInfo() {}
+
+  public String getTableName() {
+    return name;
+  }
+
+  // Include database name and table name
+  public String getTableIdentifier() {
+    if (StringUtils.isNotBlank(database)) {
+      return String.join(".", database, name);
+    } else {
+      return name;
+    }
+  }
+
+  @Data
+  public static class SparkColumnInfo {
+    private String name;
+    private String type;
+    private String comment;
+
+    private SparkColumnInfo(String name, String type, String comment) {
+      this.name = name;
+      this.type = type;
+      this.comment = comment;
+    }
+
+    public static SparkColumnInfo of(String name, String type) {
+      return of(name, type, null);
+    }
+
+    public static SparkColumnInfo of(String name, String type, String comment) {
+      return new SparkColumnInfo(name, type, comment);
+    }
+  }
+
+  public enum ParseStage {
```

Comment:
What's the meaning of `ParseStage`? Is it a standard term?

Reply:
I use something like state machine to parse SparkSQL result,  `ParseStage` is used to check which state or stage is current on.